db/compaction/compaction_job.cc

```diff
@@ -1542,7 +1542,8 @@ Status CompactionJob::OpenCompactionOutputFile(
                            &syncpoint_arg);
 #endif
   Status s;
-  IOStatus io_s = NewWritableFile(fs_, fname, &writable_file, file_options_);
```

Comment:
Is it possible that we change the fs_ pointer instead of using db_options_.fs.GetFileSystem().get() directly. It might cause potential that using different fs_ in the same compaction job.